aptible/database.go

```diff
@@ -45,75 +46,89 @@ func (c *Client) CreateDatabase(env_id int64, attrs DBCreateAttrs) (*models.Inli
 	if err != nil {
 		return nil, err
 	}
-
+	// waits for provision operation to finish
 	op_id := *op_resp.Payload.ID
 	err = c.WaitForOperation(op_id)
 	if err != nil {
 		return nil, err
 	}
-
-	payload, _, err := c.GetDatabase(db_id)
+	// gets database
+	payload, _, err := c.GetDatabaseFromHandle(env_id, *attrs.Handle)
 	return payload, nil
 }
 
-func (c *Client) GetDatabase(db_id int64) (*models.InlineResponse2014, bool, error) {
+func (c *Client) GetDatabase(db_id int64) (DBUpdates, bool, error) {
+	deleted := false
+	updates := DBUpdates{}
 	params := operations.NewGetDatabasesIDParams().WithID(db_id)
 	resp, err := c.Client.Operations.GetDatabasesID(params, c.Token)
 	if err != nil {
 		switch err.(type) {
 		case *operations.GetDatabasesIDDefault:
-			err_struct := err.(*operations.GetDatabasesIDDefault)
-			switch err_struct.Code() {
-			case 404:
-				// If deleted == true, then the database needs to be removed from Terraform.
-				return nil, true, nil
-			case 401:
-				e := fmt.Errorf("Make sure you have the correct auth token.")
-				return nil, false, e
-			default:
-				e := fmt.Errorf("There was an error when completing the request to get the database. \n[ERROR] -%s", err)
-				return nil, false, e
+			if err.(*operations.GetDatabasesIDDefault).Code() == 404 {
+				deleted = true
 			}
+			return updates, deleted, err
 		default:
-			e := fmt.Errorf("There was an unknown error. \n[ERROR] -%s", err)
-			return nil, false, e
+			return updates, deleted, err
 		}
 	}
-	return resp.Payload, false, nil
+	// get updates to container size
+	serv_href := resp.Payload.Links.Service.Href.String()
```

Comment:
Since we're doing some non-trivial parsing of the response payload here, this may be a good candidate to pull out into a private helper function and write some tests around. What do you think?

Reply:
Maybe something like `GetIDFromHref()`, since we do a similar thing with the disk size below. It's also the same for getting the service ID for db endpoints.

Reply:
Yeah that sounds reasonable. you can either do a couple smaller helpers or one one larger one for the whole thing. The former is better if you think you'll need to reuse the parsing anyplace else, otherwise one for the whole section would be fine if that's easier.